rasa/shared/core/domain.py

```diff
@@ -1954,6 +1956,36 @@ def _validate_forms(forms: Union[Dict, List], domain_slots: Dict[Text, Any]) ->
                     f"mapped in domain slots."
                 )
 
+        all_form_slots = [
+            required_slots[REQUIRED_SLOTS_KEY] for required_slots in forms.values()
+        ]
+        all_required_slots = set(chain.from_iterable(all_form_slots))
+
+        for slot, properties in domain_slots.items():
+            for mapping in properties["mappings"]:
+
+                if "conditions" in mapping:
+                    for condition in mapping["conditions"]:
```

Comment:
nitpick: that would drop one level of indent, wouldn't it?
```suggestion
                for condition in mapping.get("conditions", []):
```